### PR TITLE
ItR-14: render list or your repo in the modal window

### PR DESCRIPTION
**Pull request by MilaKolasava**
image
image

### CODE REVIEW

File: src/components/ModalWindow/ModalWindow.tsx [hunk 1]
```diff
@@ -1,20 +1,69 @@
...
+import { useEffect } from "react";
+import { ReactComponent as Cross } from "../../assets/cross.svg";
 
 interface ModalWindowProps {
   setModalActive: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
+type searchResalt = Array<searchRepoType>;
+
+type searchRepoType = {
```
Comment (glebpigulevsky) [inline src/components/ModalWindow/ModalWindow.tsx]: capitalized first letter for types or interfaces

Comment (glebpigulevsky) [reply src/components/ModalWindow/ModalWindow.tsx]: Repository

Comment (MilaKolasava) [reply src/components/ModalWindow/ModalWindow.tsx]: DONE

File: src/components/ModalWindow/ModalWindow.tsx [hunk 2]
```diff
@@ -1,20 +1,76 @@
...
+        <div className="modal-header-wrapper">
+          <div className="modal-header">
+            <Cross
+              className="modalCross"
+              onClick={() => {
+                props.setModalActive(false);
+              }}
+            />
+            <h2 className="modal-header">
+              {" "}
```
Comment (glebpigulevsky) [inline src/components/ModalWindow/ModalWindow.tsx]: is it neccesary?

File: src/components/ModalWindow/ModalWindow.tsx [hunk 3]
```diff
@@ -1,20 +1,76 @@
...
+                setSelectedRepo(repo);
+              }}
+            >
+              <b>{repo.name}</b> <br />
+              <a href={repo.html_url} target="_blank">
+                {repo.html_url}
+              </a>
+            </li>
+          ))}
+        </ol>
```
Comment (glebpigulevsky) [inline src/components/ModalWindow/ModalWindow.tsx]: I prefer to move it in the separate component

File: src/components/ModalList/ModalList.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    axios
+      .get<SearchResult>("https://api.github.com/users/MilaKolasava/repos")
+      .then((response) => {
+        setRepositories(response.data);
+      });
+  }, []);
+
+  return (
+    <ol>
+      {repositories.map((repo) => (
```
Comment (glebpigulevsky) [inline src/components/ModalList/ModalList.tsx]: repository

File: src/components/ModalList/ModalList.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+type Repository = {
+  name: string;
+  id: number;
+  html_url: string;
+};
+
+function ModalList() {
+  const [repositories, setRepositories] = useState<Repository[]>([]);
+  const [selectedRepo, setSelectedRepo] = useState<Repository | null>(null);
```
Comment (glebpigulevsky) [inline src/components/ModalList/ModalList.tsx]: selectedRepository

File: src/components/ModalList/ModalList.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+        <li
+          key={repo.id}
+          className={selectedRepo === repo ? "selected" : ""}
+          onClick={() => {
+            setSelectedRepo(repo);
+          }}
+        >
+          <b>{repo.name}</b> <br />
+          <a href={repo.html_url} target="_blank">
+            {repo.html_url}
```
Comment (glebpigulevsky) [inline src/components/ModalList/ModalList.tsx]: repository

File: src/components/ModalWindow/ModalWindow.tsx (hunk 3 above)
Comment (MilaKolasava) [reply src/components/ModalWindow/ModalWindow.tsx]: done

File: src/components/ModalList/ModalList.tsx (hunk 4 above)
Comment (MilaKolasava) [reply src/components/ModalList/ModalList.tsx]: ok

File: src/components/ModalList/ModalList.tsx (hunk 5 above)
Comment (MilaKolasava) [reply src/components/ModalList/ModalList.tsx]: ok

File: src/components/ModalList/ModalList.tsx (hunk 6 above)
Comment (MilaKolasava) [reply src/components/ModalList/ModalList.tsx]: ok

File: src/components/ModalWindow/ModalWindow.tsx (hunk 2 above)
Comment (MilaKolasava) [reply src/components/ModalWindow/ModalWindow.tsx]: no

**Review by glebpigulevsky — APPROVED** [review]
Красота

**Pull request merged**